feature_engine/selection/drop_high_PSI_features.py

```diff
@@ -0,0 +1,460 @@
+from typing import List, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.dataframe_checks import _check_contains_na, _is_dataframe
+from feature_engine.discretisation.equal_frequency import EqualFrequencyDiscretiser
+from feature_engine.discretisation.equal_width import EqualWidthDiscretiser
+from feature_engine.selection.base_selector import BaseSelector
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+Variables = Union[None, int, str, List[Union[str, int]]]
+
+
+class DropHighPSIFeatures(BaseSelector):
+    """
+    DropHighPSIFeatures drops features with a Population Stability Index (PSI) above
+    a given threshold. The PSI of a feature is an indication of the shift in its
+    distribution; a feature with high PSI might therefore be seen as instable.
+
+    In fields like Credit Risk Modelling, the elimination of features with high PSI
+    is frequent and usually required by the Regulator.
+
+    The calculation of the PSI requires to compare two distributions.
+    In DropHighPSIFeatures two approaches are implemented though the
+    "basis" argument.
+
+    - If it is a pandas.DataFrame, the class will compare the distributions of the
+    X dataframe (argument of the fit method) and "basis". The two dataframes must
+    contain the same features (i.e. labels).
+
+    - If it is a dictionary, the X matrix of the fit method is split in two according
+    to time and the distribution between the two parts are compared using the PSI.
+    The dictionary must contain the label of a column with dates and the cut-off date.
```

Comment:
The issue we would then have it that PSI is usually performed between the in-time (train) and the out-of-time dataset. So this would mean that we need to pass the "full" dataset to the sklearn pipeline which does not fit with other approaches (like dropping correlated features) that are usually done on the in-time data set.